File: Exercise_4/C++/LinkedList.cpp
#include "LinkedList.hpp"
#include <iostream>
#include <string>


LinkedList::LinkedList() {
    this->first = NULL;
    this->last = NULL;
    this->size = 0;
}


int LinkedList::getSize() {
    return this->size;
}

void LinkedList::add(Node *node) {
    if (!last) {
        first = node;
        last = first;
        size++;
        return;
    }
    last->next = node;
    last = last->next;
    size++;
}

Node* LinkedList::get(int index) {
    if (first == NULL || index >= size)
        return NULL;
    Node *node = first;
    int current = 0;
    while (current < index) {
        node = node->next;
        current++;
    }
    return node;
}

void LinkedList::deleteFirst() {
    if (first == NULL || size <= 0) {
        std::cout << "Cannot Delete from empty list" << std::endl;
        return;
    }
    Node* oldfirst = first;
    first = first->next;
    delete oldfirst;
    size--;
}

File: Exercise_4/Java/LinkedList.java
import java.util.NoSuchElementException;

/**
 * Created by milonaki on 22/12/2015.
 */
public class LinkedList {

    private Node first;
    private Node last;
    private int size;

    public int size() {
        return size;
    }

    public void add(Node node) {
        if (last == null) {
            first = node;
            last = first;
            size++;
            return;
        }
        last.next = node;
        last = last.next;
        size++;
    }

    public Node get(int index) {
        if (first == null || index >= size)
            throw new NoSuchElementException();
        Node node = first;
        int current=0;
        while (current < index) {
            node = node.next;
            current++;
        }
        return node;
    }
}

class Node {
    Node next;
}
